### ISSUE TITLE
A few System.Funcs not wrapped 

### ISSUE THREAD

**Issue by glchapman** [fsprojects/FSharp.Control.Reactive#51]
In a few cases, the functions in the Observable module end up taking System.Func parameters (rather than a corresponding F# function). I assume these are inadvertent, since in all other cases the Observable functions take F# functions as parameters (where appropriate). The cases I found are:

    longCountSatisfy (predicate: Func<'a,bool>)

    generateTimeSpan (condition: Func<'state,bool>) (iterate: Func<'state,'state>)

Also, the Observable.create function has type:
```(IObserver<'T> -> Action) -> IObservable<'T>```

It might be worth rewriting it to wrap the Action, for example:
```fsharp
    let create : (IObserver<'T> -> (unit -> unit)) -> IObservable<'T> =
        fun subscribe ->
            Observable.Create(fun observer -> Action(subscribe observer))
```

**Comment by bordoley** [comment]
https://github.com/fsprojects/FSharp.Control.Reactive/pull/56 fixes the above function. Other functions that appear to need fixing include:

generateTimeSpan
longCountSatisfy

**Comment by panesofglass** [comment]
Do you mind submitting the fix for those, as well?

**Comment by panesofglass** [comment]
I made those fixes and will release as 3.1.0.
